docs/closure-compiler.md

```diff
@@ -0,0 +1,697 @@
+components for closure.
+
+#### constants.js
+
+```js
+
+/** @enum {string} */
+export const cssClasses = {
+  FADE_IN: 'mdc-example--fade-in',
+  FADE_OUT: 'mdc-example--fade-out',
+  IMPORTANT_MSG_FLASH: 'mdc-example__important-msg--flash',
+};
+
+/** @enum {string} */
+export const strings = {
+  IMPORTANT_MSG_SELECTOR: '.mdc-example__important-msg',
+};
+
+/** @enum {number} */
+export const numbers = {
+  FADE_DURATION_MS: 3000,
+};
+```
+
+#### adapter.js
+
+```js
+/** @record */
+export default class MDCExampleAdapter {
+  /**
+   * Adds a class to the root element.
+   * @param {string} className
+   */
+  addClass(className) {}
+
+  /**
+   * Removes a class from the root element.
+   * @param {string} className
+   */
+  removeClass(className) {}
+
+  /**
+   * Registers an event listener `handler` for event type `type` on the root element.
+   * @param {string} type
+   * @param {!Function} handler
+   */
+  registerInteractionHandler(type, handler) {}
+
+  /**
+   * Un-registers an event listener `handler` for event type `type` on the root element.
+   * @param {string} type
+   * @param {!Function} handler
+   */
+  deregisterInteractionHandler(type, handler) {}
+
+  /**
+   * Adds a class to the `important-msg` element.
+   * @param {string} className
+   */
+  addClassToImportantMsg(className) {}
+
+  /**
+   * Removes a class from the `important-msg` element.
+   * @param {string} className
+   */
+  removeClassFromImportantMsg(className) {}
+}
+```
+
+#### foundation.js
+
+```js
+import MDCFoundation from '@material/base/foundation';
+import MDCExampleAdapter from './adapter';
+import {cssClasses, strings, numbers} from './constants';
+
+/** @final @extends {MDCFoundation<!MDCExampleAdapter>} */
+export default class MDCExampleFoundation extends MDCFoundation {
+  /** @return enum{cssClasses} */
+  static get cssClasses() {
+    return cssClasses;
+  }
+
+  /** @return enum{strings} */
+  static get strings() {
+    return strings;
+  }
+
+  /** @return enum{numbers} */
+  static get numbers() {
+    return numbers;
+  }
+
+  /** @return {!MDCExampleAdapter} */
+  static get defaultAdapter() {
+    return {
+      addClass: () => {},
+      removeClass: () => {},
+      registerInteractionHandler: () => {},
+      deregisterInteractionHandler: () => {},
+      addClassToImportantMsg: () => {},
+      removeClassFromImportantMsg: () => {},
+    };
+  }
+
+  /**
+   * @param {!MDCExampleAdapter} adapter
+   */
+  constructor(adapter) {
+    super(Object.assign(MDCExampleFoundation.defaultAdapter, this));
+    /** @private {boolean} */
+    this.active_ = false;
+    /** @private {number} */
+    this.fadeInTimer_ = 0;
+    /** @private {number} */
+    this.fadeOutTimer_ = 0;
+  }
+
+  /**
+   * @return {boolean}
+   */
+  isActive() {
+    return this.active_;
+  }
+
+  /**
+   * @param {boolean} active
+   */
+  setActive(active) {
+    const {FADE_IN, FADE_OUT, IMPORTANT_MSG_FLASH} = cssClasses;
```

Comment:
It is one of those "if you update this then update that". Aka if you update constants.js cssClasses, then you should update the implementation of foundation.js setActive method. It could be unintuitive to the developer to remember to update both places? Its a nit though